src/commands/containers/attachShellContainer.ts

```diff
@@ -46,7 +47,7 @@ export async function attachShellContainer(context: IActionContext, node?: Conta
             // If so use it, otherwise use sh
             try {
                 // If this succeeds, bash is present (exit code 0)
-                await execAsync(`docker exec -i ${node.containerId} sh -c "which bash"`);
+                await execAsync(`${dockerExePath()} exec -i ${node.containerId} sh -c "which bash"`);
```

Comment:
Should the Docker path be quoted to allow for paths with whitespace?

Reply:
Good question. The user can include quotes in the path as necessary, for example the following works:

```json
"docker.dockerPath": "'/Users/karolz/docker local'"
```

(the above being hard link to `/usr/local/bin/docker`)

Reply:
Definitely true, but do we expect the user to *know* that they should do that?  (We don't generally require users to self-quote their paths, for example, in the task/launch JSON properties.)

Reply:
Given that we do not expect it to be a common occurrence (that the path to Docker executable contains whitespace), I just updated the description of the setting to provide some guidance for the user.